osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -186,13 +186,8 @@ public override bool HandleRotation(float delta)
 
                 if (h is IHasPath path)
                 {
-                    var controlPoints = path.Path.ControlPoints.Select(p =>
-                        new PathControlPoint(RotatePointAroundOrigin(p.Position, Vector2.Zero, delta), p.Type)).ToArray();
-
-                    // Importantly, update as a single operation so automatic adjustment of control points to different
-                    // curve types does not unexpectedly trigger and change the slider's shape.
-                    path.Path.ControlPoints.Clear();
-                    path.Path.ControlPoints.AddRange(controlPoints);
+                    foreach (PathControlPoint t in path.Path.ControlPoints)
+                        t.Position = RotatePointAroundOrigin(t.Position, Vector2.Zero, delta);
```

Comment:
I'm slightly confused by this.. This is exactly what the code was before a fix was applied which was mentioned in the inline comment.

https://github.com/ppy/osu/pull/19865/files#diff-f9c1b963f191550071598ebfe9c4eb13303daf6924abd20933cc4d261b5ede4eL186-R195

Is there some guarantee that this can't happen anymore or something? Why would only this part of code have changed and not `HandleFlip` as well?

Reply:
This is intended to revert the previous fix and replace it. The important part is the `AddOnce` added elsewhere in this PR.

I think it's in a better place than manually handling this each time (and more importantly, manually handling leads to the noticed performance regression).

Good catch on `HandleFlip`, I've updated it too. I guess the saving grace there is `HandleFlip` is generally called once, not thousands of times per second.

Reply:
Right, it's not actually the `AddOnce` - it's the `Schedule` itself that is important.